File: datasets/C++/Claude/Algorithm/easy5.cpp
#include <iostream>
#include <vector>
using namespace std;

string gameOfStones(int n) {
    // dp[i] = true if current player can win with i stones
    vector<bool> dp(n + 1, false);
    
    // Base cases: if we can make a move and opponent will be in losing position
    for (int i = 2; i <= n; i++) {
        // Try removing 2, 3, or 5 stones
        int moves[] = {2, 3, 5};
        for (int move : moves) {
            if (i >= move && !dp[i - move]) {
                dp[i] = true;
                break;
            }
        }
    }
    
    return dp[n] ? "First" : "Second";
}

// Optimized pattern-based solution
string gameOfStonesOptimized(int n) {
    return (n % 7 == 0 || n % 7 == 1) ? "Second" : "First";
}

int main() {
    int t;
    cout << "Enter the number of test cases: ";
    cin >> t;
    
    cout << "Enter " << t << " test case(s), each containing the number of stones:" << endl;
    
    for (int i = 0; i < t; i++) {
        int n;
        cin >> n;
        
        // Use either the DP solution or optimized solution
        string result = gameOfStones(n);
        // string result = gameOfStonesOptimized(n);  // Uncomment for optimized version
        
        cout << result << endl;
    }
    
    return 0;
}

File: datasets/Python/Claude/Algorithm/easy5.py
def gameOfStones(n):
    # dp[i] = True if current player can win with i stones
    dp = [False] * (n + 1)
    
    # Base cases: if we can make a move (remove 2, 3, or 5 stones)
    # and the opponent will be in a losing position, we win
    for i in range(2, n + 1):
        # Try removing 2, 3, or 5 stones
        moves = [2, 3, 5]
        for move in moves:
            if i >= move and not dp[i - move]:
                dp[i] = True
                break
    
    return "First" if dp[n] else "Second"

def main():
    print("Enter the number of test cases:")
    t = int(input())
    
    print(f"Enter {t} test case(s), each containing the number of stones:")
    
    for i in range(t):
        n = int(input())
        result = gameOfStones(n)
        print(result)

if __name__ == "__main__":
    main()